valen-ui/src/main/java/be/twofold/valen/ui/viewers/TextViewer.java

```diff
@@ -4,17 +4,23 @@
 import javafx.scene.*;
 import javafx.scene.control.*;
 
+import java.io.*;
+import java.nio.*;
+import java.nio.charset.*;
+
 public class TextViewer extends TextArea implements Viewer {
     @Override
-    public boolean canPreview(AssetType type) {
-        return type == AssetType.Text;
+    public boolean canPreview(Asset asset) {
+        return asset.tags().contains(AssetTypeTags.Text);
     }
 
     @Override
-    public void setData(Object data) {
-        if (data instanceof String) {
-            setText((String) data);
-        }
+    public boolean setData(Asset asset, Archive archive) throws IOException {
+        ByteBuffer data = archive.loadRawAsset(asset.id());
+        byte[] bytes = new byte[data.remaining()];
+        data.get(bytes);
```

Comment:
Isn't there like a helper method to read a byte array without a temp variable?

Reply:
Better implementation will be added in next PR